src/main/java/com/exasol/adapter/jdbc/JDBCAdapter.java

```diff
@@ -151,6 +164,22 @@ private boolean requiresRefreshOfVirtualSchema(final Map<String, String> propert
                 || AdapterProperties.isRefreshingVirtualSchemaRequired(properties);
     }
 
+    /**
+     * Create or get existing {@link RemoteConnectionFactory} instance.
+     *
+     * @param metadata metadata to use
+     * @param properties adapter properties
+     * @return connection factory
+     */
+    protected RemoteConnectionFactory getOrCreateConnectionFactory(final ExaMetadata metadata,
+              final AdapterProperties properties) {
+        // Open question: can metadata and properties be changed during connection lifetime?
+        //  If yes, our connection factory is implemented wrongly.
+        if (this.connectionFactory == null)
+            this.connectionFactory = new RemoteConnectionFactory(metadata, properties);
```

Comment:
```suggestion
        if (this.connectionFactory == null) {
            this.connectionFactory = new RemoteConnectionFactory(metadata, properties);
        }
```
Our coding convention is to always use blocks.